boards/common/atmega: Refactoring periph_conf.h

maribu opened the pull request:
### Contribution description

- Moved code for `periph_conf` of all ATmega based boards to `boards/common/atmega`
- Added possibility to override config from individual board:
  - Named file `periph_conf_atmega_common.h` and let this be included from  `board/$BOARD/include/periph_conf.h` to allow modifications
  - Guarded individual periph configs by `#ifndef $PERIPH_NUMOF` ... `#endif`

### Testing procedure

Do basic testing on ATmega based boards

- [x] Arduino Duemilanove (@maribu, (worked on an Arudino UNO which also uses the ATmega328p))
- [x] Arduino Mega2560 (@smlng)
- [x] Arduino Uno (@smlng)
- [ ] Jiminy Mega 256rfr2
- [ ] Mega1284P-Xplained
- [ ] Waspmote PRO v1.2

### Issues/PRs references

None

----

Update 1: Added testing state

maribu commented:
Let's ask Murdock how much code this PR broke :-)

maribu commented:
Please wait with the review for now. I'd like to fix any issues detected by the CI first

maribu commented:
@kYc0o, @aabadie, @smlng: This PR is now ready for review. (I will also stop force pushing without asking now to not interfere with the review ;-))

smlng commented:
another ACK would be nice

smlng commented:
tested with Arduino-Uno, still working too.

kYc0o commented:
I'd ACK it but unfortunately I don't have any of these boards anymore. Code-wise it's OK for me.

maribu commented:
I added the current testing state to the PR description. I also found a duplicated "USEMODULE" for the Arduino Duemilanove (there was an identical one in `Makefile.dep`, so I deleted the duplicate in `Makefile.include`).

I added the `needs >1 ACK`. I think this PR is just the kind of change which easily breaks stuffs. So having good testing and - as @smlng suggest - a second ACK would be good thing :-)

maribu commented:
@miri64: You're the single author in the Arduino Duemilanova's `board.h`, do you have still such a board at hand and mind to test?

@ZetaR60: Could you test on the Mega1284P-Xplained?

@aabadie: Do you have a Waspmote Pro at hand and could test this?

@josar: Could you test this PR for the Jiminy board?

maribu commented:
OK, I tested compiling for the Arduino Duemilanove and flashed the result successfully on an Arduino UNO. (Except for the different baud rate while flashing the Arduino UNO and the Arduino Duemilanove are identical.) So only the Mega Xplained, the Waspmote Pro and the Jiminy need to be tested

smlng commented:
please squash, I'm willing to get this in even without having tested all boards, which is likely just a problem of having them.

maribu commented:
> please squash

done :-)
